src/go/MONIT/intelligence.go

```diff
@@ -0,0 +1,320 @@
+var allAlertData []alertData
+
+type matchers struct {
+	Name  string `json:"name"`
+	Value string `json:"value"`
+}
+
+type silenceData struct {
+	Matchers  []matchers `json:"matchers"`
+	StartsAt  time.Time  `json:"startsAt"`
+	EndsAt    time.Time  `json:"endsAt"`
+	CreatedBy string     `json:"createdBy"`
+	Comment   string     `json:"comment"`
+}
+
+type config struct {
+	CMSMONURL      string        `json:"cmsmonURL"`
+	GetAlertsAPI   string        `json:"getAlertsAPI"`
+	PostSilenceAPI string        `json:"postSilenceAPI"`
+	HttpTimeout    int           `json:"httpTimeout"`
+	Interval       time.Duration `json:"interval"`
+	CreatedBy      string        `json:"createdBy"`
+	SeverityFilter string        `json:"severityFilter"`
+	SearchingLabel string        `json:"searchingLabel"`
+	UniqueLabel    string        `json:"uniqueLabel"`
+	Comment        string        `json:"comment"`
+	Verbose        int           `json:"verbose"`
+}
+
+var configJSON config
+
+//-------STRUCTS---------
+
+//function for get request on /api/v1/alerts alertmanager endpoint for fetching alerts.
+func get(data interface{}) {
+
+	//GET API for fetching only GGUS alerts.
+	apiurl := configJSON.CMSMONURL + configJSON.GetAlertsAPI
+
+	req, err := http.NewRequest("GET", apiurl, nil)
+	req.Header.Add("Accept-Encoding", "identity")
+	req.Header.Add("Accept", "application/json")
+
+	timeout := time.Duration(configJSON.HttpTimeout) * time.Second
+	client := &http.Client{Timeout: timeout}
+
+	if configJSON.Verbose > 1 {
+		dump, err := httputil.DumpRequestOut(req, true)
+		if err == nil {
+			log.Println("Request: ", string(dump))
+		}
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		panic(err)
+	}
+	defer resp.Body.Close()
+
+	byteValue, err := ioutil.ReadAll(resp.Body)
+
+	if err != nil {
+		log.Printf("Unable to read JSON Data from AlertManager GET API, error: %v\n", err)
+		return
+	}
+
+	err = json.Unmarshal(byteValue, &data)
+	if err != nil {
+		if configJSON.Verbose > 0 {
+			log.Println(string(byteValue))
+		}
+		log.Fatalf("Unable to parse JSON Data from AlertManager GET API, error: %v\n", err)
+	}
+
+	if configJSON.Verbose > 1 {
+		dump, err := httputil.DumpResponse(resp, true)
+		if err == nil {
+			log.Println("Response: ", string(dump))
+		}
+	}
+
+}
+
+func silence(data amJSON, mData amJSON) {
+	// GET API for fetching only GGUS alerts.
+	apiurl := configJSON.CMSMONURL + configJSON.PostSilenceAPI
+
+	var sData silenceData
+	sData.StartsAt = mData.StartsAt //Start Time equal to maintenance alert	   //So that when maintenance alerts vanishes, ongoing alerts comes back alive.
+	sData.EndsAt = mData.EndsAt     //End Time equal to maintenance alert	//So that when maintenance alerts vanishes, ongoing alerts comes back alive.
+	sData.CreatedBy = configJSON.CreatedBy
+	sData.Comment = configJSON.Comment
+	var m matchers
+
+	m.Name = configJSON.UniqueLabel
+	for k, v := range data.Labels {
+		if k == configJSON.UniqueLabel {
+			m.Value = v.(string)
```

Comment:
You need to use switch clause to properly cast data types, what if v will be nil and you're trying to cast it to string?